### PR TITLE
Add Save pasteboard content to a file feature

### PR DESCRIPTION
**Pull request by Tarektouati**
Before : 
Screenshot 2020-05-16 at 15 20 09
Screenshot 2020-05-19 at 13 17 26

After:
Screenshot 2020-05-16 at 15 20 09
Screenshot 2020-05-19 at 12 58 43

### CODE REVIEW

File: Pasteboard Viewer/ContentView.swift [hunk 1]
```diff
@@ -46,6 +46,7 @@ struct ContentView: View {
 				}
 					.labelsHidden()
 					.padding()
+				Button("Save to a file", action:{self.saveTofile(selectedType: self.selectedType)}).padding()
```
Comment (sindresorhus) [inline Pasteboard Viewer/ContentView.swift:49]: This should be in the apps main menu, not in the window.

File: Pasteboard Viewer/Utilities.swift [hunk 2]
```diff
@@ -511,6 +511,28 @@ extension View {
 	func eraseToAnyView() -> AnyView {
 		AnyView(self)
 	}
+	
+	func saveTofile(selectedType: Pasteboard.PasteboardType?) -> Void {
+		guard AppDelegate.shared.window != nil else { return  }
+		guard let fileName: String = selectedType?.title else { return }
+		guard let pasteboard: Pasteboard = selectedType?.pasteboard else { return}
+		guard let fileType: NSPasteboard.PasteboardType = selectedType?.type else { return }
```
Comment (sindresorhus) [inline Pasteboard Viewer/Utilities.swift:519]: This can be one guard statement and simplified a lot.

File: Pasteboard Viewer/Utilities.swift [hunk 3]
```diff
@@ -511,6 +511,28 @@ extension View {
...
+		panel.directoryURL = FileManager.default.homeDirectoryForCurrentUser
+		panel.nameFieldStringValue = fileName
+		panel.beginSheetModal(for: AppDelegate.shared.window){ (result) in
+			if result.rawValue == NSApplication.ModalResponse.OK.rawValue,
+			let url = panel.url {
+			do {
+				try pasteboard.nsPasteboard.data(forType: fileType)?.write(to: url)
+			} catch {
+				print("Unable to save file")
+				print(error.localizedDescription)
```
Comment (sindresorhus) [inline Pasteboard Viewer/Utilities.swift:531]: This should use `.presentError()`

**Comment by sindresorhus** [comment]
Thanks for working on this, but this PR needs more work.

File: Pasteboard Viewer/Utilities.swift [hunk 4]
```diff
@@ -511,6 +511,28 @@ extension View {
...
 	}
+	
+	func saveTofile(selectedType: Pasteboard.PasteboardType?) -> Void {
+		guard AppDelegate.shared.window != nil else { return  }
+		guard let fileName: String = selectedType?.title else { return }
+		guard let pasteboard: Pasteboard = selectedType?.pasteboard else { return}
+		guard let fileType: NSPasteboard.PasteboardType = selectedType?.type else { return }
+		
+		let panel = NSSavePanel()
+		panel.directoryURL = FileManager.default.homeDirectoryForCurrentUser
```
Comment (sindresorhus) [inline Pasteboard Viewer/Utilities.swift:522]: I don't see why the home directory would be the default. Better to not set it.

File: Pasteboard Viewer/ContentView.swift (hunk 1 above)
Comment (sindresorhus) [reply Pasteboard Viewer/ContentView.swift:49]: From the original issue:

> We could add a "Save" menu item for that.

Comment (Tarektouati) [reply Pasteboard Viewer/ContentView.swift:49]: Right, didn't understand that, I thought it was:  add save button the side menu, I'll change that

**Comment by sindresorhus** [comment]
Please follow the macOS Human Interface Guidelines for the "File" menu.

**Comment by sindresorhus** [comment]
The menu item should be "Save As…" as it's not saving directly.

File: Pasteboard Viewer/ContentView.swift [hunk 5]
```diff
@@ -28,9 +28,14 @@ struct ContentView: View {
 	private func setWindowTitle() {
 		AppDelegate.shared.window.title = selectedType?.title ?? App.name
 	}
+
```
Comment (sindresorhus) [inline Pasteboard Viewer/ContentView.swift:31]: You can make the AppDelegate `selectedType` a `Binding` instead and then only set the binding in the init here from the `@State` (`$selectedType`) and the binding will keep the value up to date itself.

**Comment by sindresorhus** [comment]
I'm going to close this for lack of activity and also because it's simply too far from being mergeable. Also, macOS 11, will bring SwiftUI improvements that will make it much easier to achieve this, so I prefer to wait.

**Pull request closed without merging**